File: BookRent/Model/Book.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace BookRent
{
    [Serializable]
    public class Book : ModelBase
    {
        public string ISBN { get; set; }
        public string Name { get; set; }
        /// <summary>
        /// 入库日期
        /// </summary>
        public DateTime InDate { get; set; }
        /// <summary>
        /// 价格
        /// </summary>
        public double Price { get; set; }
        /// <summary>
        /// 拼音首字母
        /// </summary>
        public string Pinyin { get; set; }
        /// <summary>
        /// 购买方式
        /// </summary>
        public string BuyFrom { get; set; }
        /// <summary>
        /// 备注
        /// </summary>
        public string Remark { get; set; }
        /// <summary>
        /// 出版社
        /// </summary>
        public string Publisher { get; set; }
        /// <summary>
        /// 作者
        /// </summary>
        public string Author { get; set; }

        /// <summary>
        /// 入库数量
        /// </summary>
        public int TotalCount { get; set; }

        private int availableCount;
        /// <summary>
        /// 现有数量
        /// </summary>
        public int AvailableCount
        {
            get
            {
                return availableCount;
            }
            set
            {
                if (value < 0 || value > TotalCount)
                {
                    throw new ArgumentOutOfRangeException("AvailableCount");
                }

                availableCount = value;
            }
        }        

        public override string ToString()
        {
            return string.Format("{0} {1}", Pinyin, Name);
        }
    }

    public enum IsbnAction
    {
        Request = 0,
        Response = 1
    }
}

File: BookRent.Web/server/book.js
var _ = require('underscore')
var fake = require('../util/data_generator');
var books = [];

module.exports = function(app){
    app.get('/book/list', function(req, res){
        res.end(JSON.stringify(books));
    });
    app.get('/book/:id', function(req, res){
        var id = req.params.id;
        var book = _.findWhere(books, {id: parseInt(id)});
        res.end(JSON.stringify(book));
    });
    app.post('/book/save', function(req, res){
        var book = req.body;
        if(null == book) return;

        if(null == book.id){
            book.id = books.length;
            books.push(book);
        }else{
            var index = _.findIndex(books, {id: book.id});
            if(-1 != index){
                books[index] = book;
            }
        }
        res.end(book.id.toString());
    });
    app.delete('/book/:id', function(req, res){
        console.log('delete book: id = '+req.params.id);
        var index = _.findIndex(books, {id: parseInt(req.params.id)});
        if(-1 != index){
            books.splice(index, 1);
        }
        res.end(req.params.id);
    });
    initBooks();
};

function initBooks(){
    console.log('init books');
    for(var i=0;i<50;i++){
        books.push({
            id:i,
            isbn:fake.fakeIsbn(),
            name:'一只特立独行的猪',
            totalCount:10,
            availableCount:8,
            pinyin:'yztldxdz',
            inDate:'2017-01-07',
            price:13.5,
            buyFrom:i%3,
            publisher:'中华书局',
            author:'王小波',
            remark:'test'
        });
    }
}
